### ISSUE TITLE
Bug while authenticating Expected: AuthenticationOk, Got: BeAuthenticationSasl(["SCRAM-SHA-256", ""])

### ISSUE THREAD

**Issue by DominikBB** [giacomocavalieri/squirrel#4]
Hey, I tried to use the service but get some runtime errors.

Folder structure:
![image](https://github.com/user-attachments/assets/ed11438b-9dd0-4344-b33f-a92ae7925235)

Postgres setup:
Just a basic docker image
```
docker run --name phoenix-postgres -e POSTGRES_USER=postgres -e POSTGRES_PASSWORD=postgres -p 5500:5432 -d postgres
```

I also set up the environment variables as well.

When running gleam run -m squirrel I get:
```
Error: Cannot authenticate

I ran into an unexpected problem while trying to authenticate with the Postgres server. This is most definitely a bug!

Please open an issue at https://github.com/giacomocavalieri/squirrel/issues/new with some details about what you where doing, including the following message:
  Expected: AuthenticationOk, Got: BeAuthenticationSasl(["SCRAM-SHA-256", ""])
```

**Comment by giacomocavalieri** [comment]
Hello! Thanks so much for trying squirrel out, I've fixed that bug on main but I still have to make a release.

In a nutshell, I do not properly handle authentication so for now it just fails when the server asks for the SHA auth. If you want to try it anyway you could either
- disable password authentication for the moment, that should fix the issue at hand
- or, if you want to keep using authentication you could clone the repo and add it as a path dep in your Gleam project, this way you can experiment with it before I officially release the fix

In any case it shouldn't be long before I make a new release! Thank you for the patience 💜

**Comment by DominikBB** [comment]
Cool! I'll wait for the release :) thanks for the elaboration!

**Comment by giacomocavalieri** [comment]
I've released v1.2.0 with the SASL-SHA256 authentication fix that should close this issue, feel free to reopen it if you're still experiencing this issue! Thanks again 😊
